### PR TITLE
M0078DDG-475 New linkcheck setup at build time

### PR DESCRIPTION
**Pull request by fpinel-microej**

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -80,11 +80,11 @@
...
-    linkcheck_output = subprocess.check_output(["make", "linkcheck"], env={"PATH": os.environ["PATH"]},)
-    print(linkcheck_output.decode(errors="ignore"))
+# linkcheck_timeout = 20
+
+# trigger linkcheck on build (deprecated, setup moved to .readthedocs.yml configuration file)
+# import subprocess
+# if os.getenv("READTHEDOCS") == "True":
+#    print("RUNNING LINKCHECK")
+#    linkcheck_output = subprocess.check_output(["make", "linkcheck"], env={"PATH": os.environ["PATH"]},)
+#    print(linkcheck_output.decode(errors="ignore"))
```
Comment (tdelhomenie-microej) [inline conf.py:90]: We can removed this code IMO, it is not useful anymore.

File: .readthedocs.yml [hunk 2]
```diff
@@ -11,4 +11,14 @@ formats:
...
-  fail_on_warning: true
\ No newline at end of file
+  fail_on_warning: true
+
+build:
+  os: "ubuntu-20.04"
+  tools:
+    python: "3.10"
+  apt_packages:
+    - graphviz
```
Comment (tdelhomenie-microej) [inline .readthedocs.yml:21]: Why do we need graphviz for the linkcheck process? Does it generate some diagrams?

File: conf.py [hunk 3]
```diff
@@ -80,11 +80,11 @@
 # ignoring Github links with anchors at linkcheck
 # linkcheck_ignore = [r'https?:\/\/github\.com\/.+#.+']
 
-linkcheck_timeout = 20
```
Comment (tdelhomenie-microej) [inline conf.py:83]: Can we add this parameter back? It caused some issues in the past.

**Review by tdelhomenie-microej — CHANGES_REQUESTED** [review]

File: conf.py (hunk 1 above)
Comment (gharelMicroEJ) [reply conf.py:90]: Agreed.

**Review by gharelMicroEJ — APPROVED** [review]
Pending the remarks of @tdelhomenie-microej , I conditionally approve this PR.

File: .readthedocs.yml (hunk 2 above)
Comment (fpinel-microej) [reply .readthedocs.yml:21]: (We discussed this orally but for the record :) the graphviz package needs to be specified here for the build to work with this new build config, it was done automatically before. We do use it to draw some diagrams in the Application Dev Guide's UI section.

File: conf.py (hunk 1 above)
Comment (fpinel-microej) [reply conf.py]: Done

**Review by tdelhomenie-microej — APPROVED** [review]

**Pull request merged**